stream/hls.go

```diff
@@ -58,11 +58,15 @@ func (b *HLSBuffer) WriteSegment(name string, s []byte) error {
 
 func (b *HLSBuffer) WaitAndPopPlaylist(ctx context.Context) (m3u8.MediaPlaylist, error) {
 	for {
-
+		b.lock.Lock()
 		if b.plCacheNew {
-			return b.plCache, nil
+			defer b.lock.Unlock()
+
 			b.plCacheNew = false
+			return b.plCache, nil
 		}
+		b.lock.Unlock()
```

Comment:
Why not just defer this immediately after locking? Because the 1 second sleep can cause a deadlock?